ScheduleSync/Disciplina.cs

```diff
@@ -16,9 +16,9 @@ public Disciplina(string nome, int periodo, Universidade universidade)
         {
             Nome = nome;
             Periodo = periodo;
-            HorarioDisciplina = new List<Horario>();
-            ProfessorResponsavel = new List<Professor>();
-            AlunosMatriculados = new List<Aluno>();
+            HorarioDisciplina = [];
```

Comment:
usando os novos _collection initializer_, para melhor performance

Reply:
Primeiramente, muito obrigado pela colaboração.

Segundo, poderia me explicar melhor essa sugestão, por favor?
Se eu declarar dessa forma "HorarioDisciplina = [];" estarei inicializando um List também? Porém com melhor performance, como disse...

Reply:
Foi mal, cometi 2 enganos, o nome da funcionalidade é _collection expression_ e infelizmente para listas ela n faz nenhuma otimização, no caso de uma lista vazia... Em outros casos, incluindo listas, ela pode otimizar como uma coleção é criada. Pense nisso como um _syntax suggar_ , que vc vai escrever o código e o compilador vai fazer o trabalho de otimizar isso. [Aqui](https://sharplab.io/#v2:CYLg1APgAgTADAWAFCwIzKnABFVAWAbg21wDoAZASwDsBHIpDVATgAoAiQHg3AQffYEoGyIUgD0IrAGEA9gBsZAUwDGAF0pTqWAJLVKqgIYzKAL3kAnZFQDOygDxSARgCslygHxYHjy1gC8WavIA7lhWtp4urqx8WADeqAA0MPEAzAC+gigsUQy4bAJMeRmhdk4RHk6W0nIuatTaur5YANoJSckAuhlAA===) vc consegue ver o código criado pelo compilador (lowering) quando vc usa o _collection initializer_ e o _collection expression_.

Não sei como tá seu inglês, mas pra saber sobre essa funcionalidade e outras recomendo [essa palestra](https://youtu.be/hq6uEaMYf5U?t=1030) do Mads (ele trabalha na linguagem C#)

Reply:
Entendi. Irei ver os 2 conteúdos que me mandou agora. Obrigado mais uma vez!

Reply:
Sempre às ordens (: